tests/hikari/impl/test_rest.py

```diff
@@ -2668,6 +2669,31 @@ async def test_delete_messages_with_iterable(self, rest_client):
             ]
         )
 
+    async def test_delete_messages_with_async_iterable(self, rest_client):
+        channel = StubModel(54123)
+        iterator = iterators.FlatLazyIterator(StubModel(i) for i in range(103))
+
+        rest_client._request = mock.AsyncMock()
+
+        await rest_client.delete_messages(channel, iterator)
+
+        rest_client._request.assert_has_awaits(
+            [
+                mock.call(
+                    routes.POST_DELETE_CHANNEL_MESSAGES_BULK.compile(channel=channel),
+                    json={"messages": [str(i) for i in range(100)]},
+                ),
+                mock.call(
+                    routes.POST_DELETE_CHANNEL_MESSAGES_BULK.compile(channel=channel),
+                    json={"messages": ["100", "101", "102"]},
+                ),
+            ]
+        )
+
+    async def test_delete_messages_with_async_iterable_and_args(self, rest_client):
+        with pytest.raises(TypeError):
```

Comment:
Should match the error string to make sure the error raised is the one you expect